include/cutlass/gemm/warp/mma_mixed_input_tensor_op.h

```diff
@@ -122,8 +123,10 @@ struct FragmentShuffler <ElementMma_, ElementLoad_,
                          NumElementsInWarpFragment, 
                          NumElementsInMmaFragment,
                          Operand::kA,
-                         typename platform::enable_if<(sizeof_bits<ElementMma_>::value == 16) &&
-                                                 (sizeof_bits<ElementLoad_>::value == 8)>::type> {
+                         typename platform::enable_if<((sizeof_bits<ElementMma_>::value == 16) &&
+                                                       (sizeof_bits<ElementLoad_>::value == 8)) ||
+                                                      ((sizeof_bits<ElementMma_>::value == 8) &&
+                                                       (sizeof_bits<ElementLoad_>::value == 4))>::type> {
```

Comment:
It is pretty pleasing to see that the same `FragmentShuffler` that worked for `F16`x`S8` is working for `S8`x`S4`. I have been trying to visualize it in my head. The invariant here is the ratio `bits(ElementMma_) : bits(ElementLoad_) = 2`. Additionally, `16-by-8-by-128b` Tensor Core operation stays fixed. 

For instance, `F16` x `S4` would need a different FragmentShuffler implementation which has different ratio `16-by-8-by-128b`. correct? 

Thinking out loud, This ratio maybe useful to create a single `FragmentShuffler` implementation that will work for `F16` x `S4` as well. 

cc: @hwu36 , @IonThruster

Reply:
Yes, the shuffler is reused from `F16xS8`, namely in both cases, for each 32 bits loaded into a thread registers through `ldmatrix`, two groups of 16 bits are shuffled between threads.  Alike for `F16xS4`, the shuffler that is added into #1190 could be reused for other combinations of operands where the bitwidth of wider operand is 4x the bitwidth of the other one.

(While we're at it: it would be great if #1190 could be revisited between you, @hwu36 and myself - the changes for the case when one operand is 4x wider than the other one are rather invasive, but these combinations of operands are still deemed quite important for PyTorch, and I'm willing to work to find an approach that would be acceptable for merging into CUTLASS.)

Reply:
Yeah, the last time went in looking deep into #1190 it had mainloop changes. Those changes will affect almost all the Ampere kernels. While this is fresh, for me, let me leave a comment on #1190.